python/lsst/dax/data_generator/columns.py

```diff
@@ -517,3 +514,150 @@ def __call__(self, box, length, seed, prereq_row=None, prereq_tables=None, uniqu
 
         return (out_objectIds, out_ccdVisitIds, psFlux, psFluxSigma, flags)
 
+
+class SourceGenerator(ColumnGenerator):
+    """Class to generate Source columns from Object and Visit tables.
+
+    Parameters
+    ----------
+    filters : str
+        String where each character is a valid filter id. These
+        should probably match the values given to FilterGenerator.
+    visit_radius : float
+        Distance from the visit center within which and object is
+        considered part of that visit.
+    column_seed : int
+        Arbitrary integer that should be different from other
+        column values. Used in random number generation.
+
+    """
+
+    def __init__(self, filters="ugrizy", visit_radius=0.30, column_seed=27):
+        self.filters = filters
+        self.visit_radius = visit_radius
+        self.column_seed = column_seed
+
+    def random_col(self, min_val, max_val, length, integer=False):
+        if (min_val > max_val):
+            tmp = min_val
+            min_val = max_val
+            max_val = min_val
+        delta_value = max_val - min_val
+        col = delta_value * np.random.rand(length) + min_val
+        if integer:
+            i_col = []
+            for f in col:
+                i_col.append(int(f))
+            col = np.array(i_col)
+        return col
+
+    def __call__(self, box, length, seed, spec_cols, prereq_row=None, prereq_tables=None, unique_box_id=0,
+                 box_center=None, edge_only=False):
+        assert prereq_tables is not None, "SourceGenerator requires the Visit table."
+        assert box_center is not None, "Must supply box center"
+
+        np.random.seed(calcSeedFrom(unique_box_id, seed, self.column_seed))
+
+        visit_table = prereq_tables['CcdVisit']
+        object_table = prereq_tables['Object']
+
+        objects_inside_box = object_table[(object_table['psRa'] >= box.raA) &
+                                          (object_table['psRa'] < box.raB) &
+                                          (object_table['psDecl'] >= box.decA) &
+                                          (object_table['psDecl'] < box.decB)]
+        v_radius = self.visit_radius * 1.5  # Go a little bit bigger so nothing is missed.
+        min_dec = box.decA - v_radius
+        max_dec = box.decB + v_radius
+        trimmed_visit = visit_table.loc[(visit_table['decl'] >= min_dec) & (visit_table['decl'] <= max_dec)]
```

Comment:
This usage of`.loc[...]` is the same as `visit_table[...]` (but not all usage of `.loc` works that way). A word of caution that this is a view, not a copy, but I don't think that affects anything here.